scripts/hash-exec-copy.py

```diff
@@ -1,104 +1,197 @@
+def parse_arguments():
+    """Parse command line arguments.
+
+    Returns:
+        datetime: The parsed date object
+    """
+    parser = argparse.ArgumentParser(
+        description="Fetch an executive vote document and calculate its keccak hash")
+    parser.add_argument(
+        "date",
+        help=f"Date to find executive copy for (format: {INPUT_DATE_FORMAT})"
+    )
+
+    args = parser.parse_args()
+    date_string = args.date.replace("date=", "")
 
     try:
-        date = datetime.strptime(date_string, INPUT_DATE_FORMAT)
+        return datetime.strptime(date_string, INPUT_DATE_FORMAT)
     except ValueError:
-        raise SystemExit(f"Invalid date format. Please use {INPUT_DATE_FORMAT}.")
-
-    # Find the executive doc for the given date
-    POSSIBLE_EXEC_TITLES = [
-        f"Executive%20vote%20-%20{date.strftime('%B %d, %Y')}.md",
-        f"Executive%20Vote%20-%20{date.strftime('%B %d, %Y')}.md",
-        # Remove leading 0 in day (e.g. 01 -> 1)
-        # There is no case that year is 0-padded
-        f"Executive%20vote%20-%20{date.strftime('%B %d, %Y').replace(' 0', ' ')}.md",
-        f"Executive%20Vote%20-%20{date.strftime('%B %d, %Y').replace(' 0', ' ')}.md"
-    ]
-    executive_content, executive_url, commit_hash = None, None, None
-    for exec_title in POSSIBLE_EXEC_TITLES:
-        try:
-            executive_content, executive_url, commit_hash = get_executive_by_title(exec_title)
-            break
-        except ValueError as e:
-            continue
-        except requests.exceptions.RequestException as e:
-            raise SystemExit(f"HTTP Request failed: {e}")
-
-    # Check if executive content was found else exit
-    if executive_content is None:
-        raise SystemExit("Error: Executive Doc not found")
-
-    # Hash target exec copy
-    exec_hash = get_content_hash(executive_content)
-
-    # Output target exec copy hash
-    print(f"Community repo commit: {commit_hash}")
-    print(f"Raw GitHub URL: {executive_url}")
-    print(f"Exec copy hash: {exec_hash}")
-
-# Execute the main function
-main()
+        raise SystemExit(
+            f"Invalid date format. Please use {INPUT_DATE_FORMAT}.")
+
+
+def main():
+    """Main function to fetch and hash an executive vote document."""
+    # Parse the date argument
+    date = parse_arguments()
+
+    # Extract year and format date
+    year = date.strftime("%Y")
+    formatted_date = date.strftime('%Y-%m-%d')
+
+    try:
+        # Find the executive file for the given date
+        exec_title = find_exec_file_by_date(year, formatted_date)
+
+        # Get the content and metadata
+        executive_content, executive_url, commit_hash = get_executive(
+            exec_title, year)
+
+        # Calculate the hash
+        exec_hash = get_content_hash(executive_content)
+
+        # Output results
+        print(f"Community repo commit: {commit_hash}")
+        print(f"Raw GitHub URL: {executive_url}")
+        print(f"Exec copy hash: {exec_hash}")
```

Comment:
Maybe it's also helpful to return the name of the file? Since it's alerady fetched, we might as well print it?

Reply:
The name of the file is already present as the last part of the URL. I'm not sure how displaying only the file name would be useful here.

Reply:
Can't the reviewer just copy/paste the relevant part of the URL for that?